File: python3/1061.py
class Solution:
    def smallestEquivalentString(self, x: str, y: str, b: str) -> str:
        c = [_ for _ in range(26)]
        k = "abcdefghijklmnopqrstuvwxyz"

        def find(i) -> int:
            if c[i] == i:
                return c[i]
            c[i] = find(c[i])
            return c[i]

        for i in range(len(x)):
            g = find(ord(x[i]) - 97)
            h = find(ord(y[i]) - 97)
            if g < h:
                c[h] = g
            else:
                c[g] = h
        return "".join([k[find(ord(i) - 97)] for i in b])

File: dart/1061.dart
class Solution {
    int find(List<int> c, int i) {
        if (c[i] == i)
            return c[i];
        c[i] = find(c, c[i]);
        return c[i];
    }

    String smallestEquivalentString(String x, String y, String b) {
        int n = x.length, l = b.length;
        String k = "abcdefghijklmnopqrstuvwxyz";
        List<int> c = [];
        for (int i = 0; i < 26; i++)
            c.add(i);
        for (int i = 0; i < n; i++) {
            int g = find(c, k.indexOf(x[i])), h = find(c, k.indexOf(y[i]));
            if (g < h)
                c[h] = g;
            else
                c[g] = h;
        }
        String r = "";
        for (int i = 0; i < l; i++)
            r += k[find(c, k.indexOf(b[i]))];
        return r;
  }
}
